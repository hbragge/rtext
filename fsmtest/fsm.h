#ifndef _FSM_H
#define _FSM_H

#define STATE_Off 0
#define STATE_On 1

extern int state;
extern int onoff_pressed;

void sm_trigger();
void sm_switch(int new_state);

#endif
