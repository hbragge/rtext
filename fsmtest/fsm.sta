Statemachine {
    State On {
        Transition targetState: /Off, condition: "onoff_pressed == 1"
    }
    State Off {
        Transition targetState: /On, condition: "onoff_pressed == 1"
    }
}
