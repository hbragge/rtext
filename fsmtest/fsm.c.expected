#include "fsm.h"
void sm_trigger() {
   switch(state) {
   case STATE_On:
      if (onoff_pressed == 1) {
         sm_switch(STATE_Off);
      }
      break;
   case STATE_Off:
      if (onoff_pressed == 1) {
         sm_switch(STATE_On);
      }
      break;
   default:
   break;
   }
}
